tensorflow/java/src/main/java/org/tensorflow/op/OperationHelper.java

```diff
@@ -0,0 +1,147 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+package org.tensorflow.op;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.tensorflow.Operation;
+import org.tensorflow.OperationBuilder;
+import org.tensorflow.Output;
+
+/**
+ * A helper for wrapping a new operation.
+ *
+ * <p>It provides utilities for wrappers to build a single operation and collect its output(s)
+ * afterward.
+ *
+ * <pre>{@code
+ * public final class Unique extends AbstractSingleOp {
+ *    ...
+ *    public static Unique create(Scope s, Input x) {
+ *        OperationHelper unique = OperationHelper.create(s, "Unique");
```

Comment:
The reason of this helper is that it simplifies the code required to retrieve multiple outputs from some operation. The trivial case is having multiple single outputs, where like you suggested below, this simple code will do the job:
```
y = unique.operation().output(0);
idx = unique.operation().output(1);
```
But it gets complicated when output list are involved, where the length of the list is unknown at compile time. For example, collecting outputs from the `ops.parsing().ParseExample` operation ends up with code way more complex than the previous example.

Now, since the code will be generated, writing complex code might not been an issue. But the intention here is to allow users writing custom ops to benefit from those helper classes, so they can avoid this hassle.